lib/aem/aemupload.js

```diff
@@ -54,6 +55,12 @@ async function* generateAEMUploadTransferRecords(options) {
             }
             sourceUrl = fileUrl(uploadFile.filePath);
         }
+
+        if (options.supportSpecialCharacters) {
```

Comment:
I would even consider checking that `options.supportSpecialCharacters` is set to boolean true. 
```suggestion
        if (options && options.supportSpecialCharacters === true) {
```

Reply:
`options` is required. It would be good to check this at the start of the function, but if options is not provided it'll fail earlier.

Reply:
I added a check at the start of the function to ensure `options` is present, otherwise an `IllegalArgumentError` is thrown.

Taylor